File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {

        let seen = new Map();

        for(let i = 0; i < nums.length; i++)
        {
            let current = nums[i];
            let lookingFor = target - current;
            
            if(seen.has(lookingFor)) {return [seen.get(lookingFor), i];}

            if(!seen.has(current)) {seen.set(current, i);}
        }
        return [0,0];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int start = 0;
        int end = numbers.Length - 1;

        while (start < end)
        {
            int val = numbers[start] + numbers[end];
            if(val == target) {return [start + 1, end + 1];}

            if (val < target) {start++;}
            else {end--;}
        }

        return [0,0];
    }
}
